### ISSUE TITLE
Suggestion: clean up the instances

### ISSUE THREAD

**Issue by tiotrom** [SimonBrazell/privacy-redirect#156]
I love this plugin and we use it by default in our TROMjaro linux. I suggest a few clean-up ideas.

**Try to only use "instances" for resilience and consistency.** 

I love when Twitter redirects to any random Nitter. All seem to look quite similar so I don't have to bother with that. I feels like a natural redirect. I have the same feeling with Invidious and Bibliogram. OpenStreetMap is the only one that is not an "instance", but it works differently so all is fine.

Reddit uses a few instances and other websites. I suggest using only the instances. Like libreddit. Works great and is consistent in terms of functionality and design.

As for search engines I suggest only leaving SearX, since there are so many instances of SearX and all work very well. They are based on the same principles of Bibliogram, Nitter, or Invidious.

Twitter to Nitter instances
Youtube to Invidious instances
Instagram to Bibliogram instances
Maps to OpenStreetMaps
Reddit to Libreddit/Teddit
Google to SearX (I also suggest redirecting most search engines to SearX, not just google. Like yahoo, bung, duckduckgo, startpage, etc.)

This way there will be a consistency and resilience built into this addon.

And for the above, would be better to use their official lists of instances to pull them at random from, rather than adding them manually in the addon code itself.

**Nitter instances list:** https://github.com/zedeus/nitter/wiki/Instances
**Bibliogram instances list:** https://git.sr.ht/~cadence/bibliogram-docs/tree/master/docs/Instances.md
**Invidious instances list:** https://github.com/iv-org/documentation/blob/master/Invidious-Instances.md#list-of-public-invidious-instances-sorted-from-oldest-to-newest
**Libreddit instances list:** https://github.com/spikecodes/libreddit#instances
**SearX instances list:** https://searx.space/

**Comment by miku86** [comment]
> And for the above, would be better to use their official lists of instances to pull them at random from, rather than adding them manually in the addon code itself.

This is what I thought about too. This would remove the manual labor of keeping the lists up-to-date in the extension. E.g. Invidious Instances has a simple [API](https://instances.invidio.us/instances.json).

We could also add a fallback, e.g. from the [web archive](https://web.archive.org/web/20201205190052/https://instances.invidio.us/instances.json).

Let us know if you are open for PRs.

**Comment by Mennaruuk** [comment]
Awesome idea!! It’d save so much time tinkering around instances as uptimes change. Configuration settings would be a challenge, though, so I’d have to go to most of these instances and “sync” my settings. Otherwise, a wonderful suggestion, and I think it’d save Mr. Brazell the headache of constantly removing and adding new instances.

**Comment by SimonBrazell** [comment]
> Let us know if you are open for PRs.

Certainly! I'm limiting the amount of time I spend on this project lately as my other paid work is piling up and deadlines are looming... so any help would be much appreciated!

**Comment by DanklyTuned** [comment]
There is a searx instance api at https://searx.space/data/instances.json if you'd like to use that instead of specific searx instances.

**Comment by xanoni** [comment]
> There is a searx instance api at https://searx.space/data/instances.json if you'd like to use that instead of specific searx instances.

This would be cool together with the randomization / pool feature.
